src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs

```diff
@@ -0,0 +1,177 @@
+                newProperties == this.Properties &&
+                newTags == this.Tags &&
+                newRules == this.Rules &&
+                newDisplayTextPrefix == this.DisplayTextPrefix &&
+                newDisplayTextSuffix == this.DisplayTextSuffix &&
+                newInlineDescription == this.InlineDescription)
+            {
+                return this;
+            }
+
+            return new TypeImportCompletionItem(
+                displayText: newDisplayText,
+                filterText: newFilterText,
+                span: newSpan,
+                sortText: newSortText,
+                properties: newProperties,
+                tags: newTags,
+                rules: newRules,
+                displayTextPrefix: newDisplayTextPrefix,
+                displayTextSuffix: newDisplayTextSuffix,
+                inlineDescription: newInlineDescription,
+                typeArity: this.TypeArity,
+                containingNamespace: this.ContainingNamespace,
+                overloadCount: this.OverloadCount);
+        }
+
+        private TypeImportCompletionItem(
+            string displayText,
+            string filterText,
+            string sortText,
+            TextSpan span,
+            ImmutableDictionary<string, string> properties,
+            ImmutableArray<string> tags,
+            CompletionItemRules rules,
+            string displayTextPrefix,
+            string displayTextSuffix,
+            string inlineDescription,
+            int typeArity,
+            string containingNamespace,
+            int overloadCount)
+            : base(displayText: displayText, filterText: filterText, sortText: sortText, span: span,
+                   properties: properties, tags: tags, rules: rules, displayTextPrefix: displayTextPrefix,
+                   displayTextSuffix: displayTextSuffix, inlineDescription: inlineDescription)
+        {
+            TypeArity = typeArity;
+            ContainingNamespace = containingNamespace;
+            OverloadCount = overloadCount;
+        }
+
+        private TypeImportCompletionItem(
+            string displayText,
+            string displayTextSuffix,
+            Glyph glyph,
+            int typeArity,
+            string containingNamespace,
+            string inlineDescription = null,
+            int overloadCount = 0)
+            : this(
+                displayText: displayText,
+                filterText: displayText,
+                sortText: displayText,
+                span: default,
+                properties: null,
+                tags: GlyphTags.GetTags(glyph),
+                rules: CompletionItemRules.Default,
+                displayTextPrefix: null,
+                displayTextSuffix: displayTextSuffix,
+                inlineDescription: inlineDescription,
+                typeArity: typeArity,
+                containingNamespace: containingNamespace,
+                overloadCount: overloadCount)
+        {
+        }
```

Comment:
why do you need all three of these constructors?

Reply:
This might be the relic from when I had to handle `ITypeDeclration`. Let me clean this up